clients/privacy-center/common/geolocation.ts

```diff
@@ -20,14 +21,15 @@ export const LOCATION_HEADERS = [
  * Lookup the "geolocation" (ie country and region) for the given request by looking for either:
  * 1) An explicit "geolocation" query param (e.g. https://privacy.example.com/some/path?geolocation=US-CA)
  * 2) Supported geolocation headers (e.g. "Cloudfront-Viewer-Country: US")
+ * 3) A geolocation API URL to infer location based on IP
  *
- * If neither of these are found, return an undefined geolocation.
+ * If none of these are found, return a null geolocation.
  *
- * NOTE: This specifically *does not* include performing a geo-IP lookup... yet!
  */
-export const getGeolocation = (
-  req: NextApiRequest
-): UserGeolocation | undefined => {
+export const lookupGeolocation = async (
+  req: NextApiRequest,
+  settings: PrivacyCenterClientSettings
+): Promise<UserGeolocation | null> => {
   // DEFER: read headers to determine & return the request's IP address
```

Comment:
@eastandwestwind this is a really important `DEFER` - without doing this, the geoIP is just going to be the IP of the privacy center